src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationOptions.cs

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Alipay.AlipayAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Alipay
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="AlipayAuthenticationHandler"/>.
+    /// </summary>
+    public class AlipayAuthenticationOptions : OAuthOptions
+    {
+        public byte[] GetPrivateKeyBytes() => Convert.FromBase64String(ClientSecret);
```

Comment:
I would just make this an implementation detail of the class that is using it, rather than putting this here.

Reply:
How about make it `internal`? In fact, I hope to optimize it if possible in the future. The problem is that `Convert.FromBase64String(ClientSecret)` each time may be a huge waste of perf because `ClientSecret` is rarely changed after once setted.
At the very original version, I wrote codes like below,
``` C#

private byte[] _privateKeyBytes;

public byte[] PrivateKeyBytes()
{
    if(_privateKeyBytes is null)
    {
        _privateKeyBytes = Convert.FromBase64String(ClientSecret);
    }
    return _privateKeyBytes;
}

```
However, `ClientSecret` is not a readonly property. So that solution is not ideal.
I thought putting this here would be a good placeholder. That's why I want to check with you.

Reply:
The `FromBase64String()` overhead is going to be pretty trivial. Also, the providers support online configuration change (e.g. from a database, Azure Key Vault etc.) as it accepts an `IOptionsMonitor<T>` instance. If the key was cached like this and only computed once, such a configuration change wouldn't take effect and would require an application restart to be applied.

As such, compute it each time it's needed as suggested.

Reply:
Fine.